### PR TITLE
update msal-lib to 1.0

### PR DESCRIPTION
**Pull request by vhvb1989**
moving the chains for msal to 1.0: https://github.com/AzureAD/microsoft-authentication-library-for-go/releases/tag/v1.0.0

### CODE REVIEW

File: cli/azd/pkg/auth/cache.go [hunk 1]
```diff
@@ -16,26 +16,28 @@ type msalCacheAdapter struct {
...
-		log.Printf("ignoring error in adapter: %v", err)
+		return err
 	}
 
-	if err := a.cache.Set(key, val); err != nil {
-		log.Printf("ignoring error in adapter: %v", err)
+	if err := a.cache.Set(cacheHints.PartitionKey, val); err != nil {
+		return err
 	}
+	return nil
```
Comment (chlowell) [inline cli/azd/pkg/auth/cache.go:40]: Nitpicky suggestion for saving some eyeball work:
```suggestion
	err = a.cache.Set(cacheHints.PartitionKey, val)
	return err
```

**Review by chlowell — APPROVED** [review]

File: cli/azd/pkg/auth/cache_test.go [hunk 2]
```diff
@@ -23,15 +25,19 @@ func TestCache(t *testing.T) {
 	}
 
 	// write some data.
-	c.Export(&d1, "d1")
-	c.Export(&d2, "d2")
+	r := c.Export(ctx, &d1, cache.ExportHints{PartitionKey: "d1"})
```
Comment (ellismg) [inline cli/azd/pkg/auth/cache_test.go:28]: NIT: Would you mind renaming `r` to `err` here?

**Review by ellismg — APPROVED** [review]
Thanks for doing this, @vhvb1989, LGTM

**Review by wbreza — APPROVED** [review]

File: cli/azd/pkg/auth/cache.go [hunk 3]
```diff
@@ -16,26 +16,25 @@ type msalCacheAdapter struct {
...
 	if err != nil {
-		log.Printf("ignoring error in adapter: %v", err)
+		return err
 	}
 
 	if err := cache.Unmarshal(val); err != nil {
-		log.Printf("ignoring error in adapter: %v", err)
+		return err
 	}
+	return nil
```
Comment (weikanglim) [inline cli/azd/pkg/auth/cache.go:28]: The old behavior here is that if `cache.Read` returned `err`, for example `os.ErrNotExist`, we would have logged the error, then continued on line 25 to perform `cache.Unmarshal` to overwrite the set value. So the change in #2259 *should* restore this behavior?

Unless that `cache.Unmarshal` has also changed underneath us to no longer ignore `nil` values.

**Pull request merged**